### PR TITLE
 refactor: JPA query tuning (RDCC-2499)

### PR DESCRIPTION
**Pull request by abhijitdiwan**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDCC-2449

### Change description ###
 refactor: JPA query tuning (RDCC-2499)

**Does this PR introduce a breaking change?** (check one with "x")

No

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -19,6 +19,7 @@ plugins {
     id 'org.springframework.boot' version '2.3.9.RELEASE'
     id "org.flywaydb.flyway" version "7.0.2"
     id 'au.com.dius.pact' version '4.1.7'
+    id 'io.freefair.lombok' version '5.3.0'
```
Comment (AdilOoze) [inline build.gradle:22]: How does this differ to the `org.projectlombok` dependency we are using?

File: src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerCreateUserWithFileUploadTest.java [hunk 2]
```diff
@@ -115,16 +124,17 @@ public void shouldUploadServiceRoleMappingsXlsxFileSuccessfully_with_extra_space
...
-                .isEqualTo(String.format("2 record(s) failed validation and 1 record(s) uploaded", 2));
+        assertThat(response).contains(entry("message", REQUEST_FAILED_FILE_UPLOAD_JSR))
+            .contains(entry("message_details",
+                format("2 record(s) failed validation and 1 record(s) uploaded", 2))).hasSize(6);
         assertThat(response.get("error_details").toString()).contains(FIRST_NAME_INVALID);
         List<CaseWorkerAudit> caseWorkerAudits = caseWorkerAuditRepository.findAll();
         assertThat(caseWorkerAudits.size()).isEqualTo(1);
-        assertThat(caseWorkerAudits.get(0).getStatus()).isEqualTo(PARTIAL_SUCCESS.getStatus());
+        assertThat(caseWorkerAudits.get(0).getStatus()).isEqualTo(PARTIAL_SUCCESS.getStatus()).hasSize(15);
+        ;
```
Comment (AdilOoze) [inline src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerCreateUserWithFileUploadTest.java:137]: Might just be GitHub's formatting but it looks like there is an additional `;` here

File: build.gradle (hunk 1 above)
Comment (abhijitdiwan) [reply build.gradle:22]: it is good to have this as we may use to check lombok generated object.

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/domain/CaseWorkerProfile.java [hunk 3]
```diff
@@ -89,7 +91,23 @@
...
+
+    @Override
+    public boolean isNew() {
+        return isNew;
+    }
+
+    public void setNew(boolean isNew) {
+        this.isNew = isNew;
+    }
 }
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/domain/CaseWorkerProfile.java:113]: @Getter , @Setter can be used ?

**Review by shreedhar-hmcts — DISMISSED** [review]

**Review by kavithagade12 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/domain/CaseWorkerProfile.java (hunk 3 above)
Comment (abhijitdiwan) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/domain/CaseWorkerProfile.java:113]: This is overridden methods Id won't smart enough to detect it and gets compilation. these are Presistable interface overrides.

**Review by shreedhar-hmcts — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerCreateUserWithFileUploadTest.java (hunk 2 above)
Comment (abhijitdiwan) [reply src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerCreateUserWithFileUploadTest.java]: done

**Review by AdilOoze — APPROVED** [review]

**Pull request merged**